azuredeploy.json

```diff
@@ -30,6 +30,67 @@
             },
             "type": "bool"
         },
+        "searchType": {
+            "defaultValue": "azure-search",
+            "allowedValues": [
+		"none-search",
+                "azure-search",
+                "elastic-search"
+            ],
+            "metadata": {
+                "description": "options of moodle global search"
+            },
+            "type": "string"
+        },
+        "azureSearchSku": {
+            "defaultValue": "basic",
+            "allowedValues": [
+		"free",
+                "basic",
+                "standard",
+                "standard2",
+                "standard3"
+            ],
+            "metadata": {
+                "description": "the search service level you want to create."
+            },
+            "type": "string"
+        },
+        "azureSearchReplicaCount": {
+            "defaultValue": 3,
+            "minValue": 1,
+            "maxValue": 12,
+            "metadata": {
+                "description": "Replicas distribute search workloads across the service. You need 2 or more to support high availability (applies to Basic and Standard only)."
+            },
+            "type": "int"
+        },
+        "azureSearchPartitionCount": {
+            "defaultValue": 1,
+	    "allowedValues": [
+		1,
```

Comment:
Indentation. There are many more below, so please go over entirely. I suppose you might be mixing tabs/spaces together in your editor environment, and you may want to unify to spaces (1 tab to 4 spaces) and look again.